GitHubBranches.ps1

```diff
@@ -154,6 +154,350 @@ filter Get-GitHubRepositoryBranch
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    [Alias('New-GitHubBranch')]
+    param(
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            Position = 1,
+            ParameterSetName = 'Uri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [string] $BranchName = 'master',
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline,
+            Position = 2)]
+        [string] $TargetBranchName,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    try
+    {
+        $getGitHubRepositoryBranchParms = @{
+            OwnerName = $OwnerName
+            RepositoryName = $RepositoryName
+            BranchName = $BranchName
+            Whatif = $false
+            Confirm = $false
+        }
+        if ($PSBoundParameters.ContainsKey('AccessToken'))
+        {
+            $getGitHubRepositoryBranchParms['AccessToken'] = $AccessToken
+        }
+        if ($PSBoundParameters.ContainsKey('NoStatus'))
+        {
+            $getGitHubRepositoryBranchParms['NoStatus'] = $NoStatus
+        }
+
+        Write-Log -Level Verbose "Getting $TargetBranchName branch for sha reference"
+        $originBranch = Get-GitHubRepositoryBranch  @getGitHubRepositoryBranchParms
+    }
+    catch
+    {
+        # Temporary code to handle current differences in exception object between PS5 and PS7
+        $throwObject = $_
+
+        if ($PSVersionTable.PSedition -eq 'Core')
+        {
+            if ($_.Exception -is [Microsoft.PowerShell.Commands.HttpResponseException] -and
+            ($_.ErrorDetails.Message | ConvertFrom-Json).message -eq 'Branch not found')
+            {
+                $throwObject = "Origin branch $BranchName not found"
+            }
+        }
+        else
+        {
+            if ($_.Exception.Message -like '*Not Found*')
+            {
+                $throwObject = "Origin branch $BranchName not found"
+            }
+        }
+
+        Write-Log -Message $throwObject -Level Error
+        throw $throwObject
+    }
+
+    $uriFragment = "repos/$OwnerName/$RepositoryName/git/refs"
+
+    $hashBody = @{
+        ref = "refs/heads/$TargetBranchName"
+        sha = $originBranch.commit.sha
```

Comment:
Technically `$originaBranch` is out-of-scope here because it was originally defined in an inner-scope (the `try` block).  Can you initialize it to `$null` outside of the `try` block to keep it in scope?

Reply:
There is no need to do this in PowerShell, but life is too short to argue. Done.